### ISSUE TITLE
[Bug]: `aws_launch_template` can't have `"aws-node-termination-handler/managed"` as a key

### ISSUE THREAD

**Issue by ba-work** [hashicorp/terraform-provider-aws#30166]
### Terraform Core Version

1.4.2

### AWS Provider Version

4.59.0

### Affected Resource(s)

* aws_launch_template

### Expected Behavior

When I add the tag `"aws-node-termination-handler/managed"` to an aws_launch_template, it gets added to the AWS resource.

### Actual Behavior

The tag doesn't get added to the AWS resource.

### Relevant Error/Panic Output Snippet

```shell
2023-03-21T15:06:35.297-0400 [WARN]  Provider "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an unexpected new value for aws_launch_template.a, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .tags_all: element "aws-node-termination-handler/managed" has vanished
      - .tags: element "aws-node-termination-handler/managed" has vanished
```

### Terraform Configuration Files

```hcl
resource "aws_launch_template" "a" { tags = { "aws-node-termination-handler/managed" = "" } }
```

### Steps to Reproduce

`terraform apply` the above config.

### Debug Output

The provider is stripping the tag and not POSTing at all:
```
2023-03-21T15:12:58.065-0400 [WARN]  Provider "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an unexpected new value for aws_launch_template.a, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .tags: element "aws-node-termination-handler/managed" has vanished
      - .tags_all: element "aws-node-termination-handler/managed" has vanished
2023-03-21T15:12:58.065-0400 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for aws_launch_template.a
2023-03-21T15:12:58.065-0400 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for aws_launch_template.a
2023-03-21T15:12:58.066-0400 [TRACE] evalApplyProvisioners: aws_launch_template.a is not freshly-created, so no provisioning is required
2023-03-21T15:12:58.066-0400 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for aws_launch_template.a
2023-03-21T15:12:58.066-0400 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for aws_launch_template.a
aws_launch_template.a: Modifications complete after 0s [id=lt-03071de98889e820a]
```

### Panic Output

N/A

### Important Factoids

Nothing important, this is minimally reproducible.

### References

[kops](https://github.com/kubernetes/kops) attempts to add this tag to the launch templates when you enable the [node-termination-handler](https://github.com/aws/aws-node-termination-handler), which is what revealed this bug. However it's easy enough to manually reproduce with a single line of Terraform.

### Would you like to implement a fix?

No

**Comment by ewbankkit** [comment]
Relates https://github.com/hashicorp/terraform-provider-aws/issues/24449.

**Comment by johnsonaj** [comment]
Resolved in #30793. Merged to `main` in #31392
